src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -93,13 +100,33 @@ private void initialize() {
      * Evicts all entries from the cache and rebuilds.
      */
     public synchronized void rebuildCache() {
+        rebuildCache(
+            KNNSettings.state().getSettingValue(KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_ENABLED),
+            KNNSettings.getCircuitBreakerLimit().getKb(),
+            KNNSettings.state().getSettingValue(KNNSettings.KNN_CACHE_ITEM_EXPIRY_ENABLED),
+            ((TimeValue) KNNSettings.state().getSettingValue(KNNSettings.KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES)).getMinutes()
+        );
+    }
+
+    /**
+     * Evict all entries from the cache and rebuilds
+     *
+     * @param isWeightLimited whether the cache should have a max weight
+     * @param maxWeight the max weight as a long. Ignored if isWeightLimited is false
+     * @param isExpirationLimited whether the cache should have a last accessed time limit
+     * @param expiryTimeInMin time an entry in cache can remain without being accessed
+     */
+    public synchronized void rebuildCache(boolean isWeightLimited, long maxWeight, boolean isExpirationLimited, long expiryTimeInMin) {
         logger.info("KNN Cache rebuilding.");
 
-        // TODO: Does this really need to be executed with an executor? Also, does invalidateAll really need to be
-        // called?
+        // TODO: Does this really need to be executed with an executor?
         executor.execute(() -> {
-            cache.invalidateAll();
-            initialize();
+            if (cache != null) {
```

Comment:
curious did we have an NPE with `cache` var or it's more of being on a safe side?

Reply:
I dont think this can be reached. The only time the cache will be null will be if rebuildCache is called before the constructor, but I dont think this can happen. I think Ill remove this.